phoenix-core/src/it/java/org/apache/phoenix/end2end/Bson3IT.java

```diff
@@ -2161,6 +2161,138 @@ public void testBsonOpsWithSqlConditionsUpdateSuccessWithTTL() throws Exception
+
+        if (expectedType.equals(PVarchar.INSTANCE.getSqlTypeName())) {
+          assertEquals("Wrong value for path " + path, expectedValue, rs.getString(1));
+        } else if (expectedType.equals(PInteger.INSTANCE.getSqlTypeName())) {
+          assertEquals("Wrong value for path " + path, ((Number) expectedValue).intValue(),
+            rs.getInt(1));
+        } else if (expectedType.equals(PLong.INSTANCE.getSqlTypeName())) {
+          assertEquals("Wrong value for path " + path, ((Number) expectedValue).longValue(),
+            rs.getLong(1));
+        } else if (expectedType.equals(PDouble.INSTANCE.getSqlTypeName())) {
+          assertEquals("Wrong value for path " + path, ((Number) expectedValue).doubleValue(),
+            rs.getDouble(1), 0.000001);
+        } else if (expectedType.equals(PBoolean.INSTANCE.getSqlTypeName())) {
+          assertEquals("Wrong value for path " + path, expectedValue, rs.getBoolean(1));
+        } else if (expectedType.equals(PBson.INSTANCE.getSqlTypeName())) {
+          assertNotNull("Null value for path " + path, rs.getObject(1));
+        }
+
+        assertFalse("Too many results for path " + path, rs.next());
+      }
+    }
+  }
+
+  @Test
+  public void testBsonValueTypeNegativeScenarios() throws Exception {
+    Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+    String tableName = generateUniqueName();
+
+    try (Connection conn = DriverManager.getConnection(getUrl(), props)) {
+      String ddl = "CREATE TABLE " + tableName + " (PK1 VARCHAR NOT NULL, C1 VARCHAR, COL BSON"
+        + " CONSTRAINT pk PRIMARY KEY(PK1)) "
+        + (this.columnEncoded ? "" : "COLUMN_ENCODED_BYTES=0");
+      conn.createStatement().execute(ddl);
+
+      String sample1 = getJsonString("json/sample_01.json");
+      BsonDocument bsonDocument1 = RawBsonDocument.parse(sample1);
+
+      PreparedStatement stmt =
+        conn.prepareStatement("UPSERT INTO " + tableName + " VALUES (?,?,?)");
+      stmt.setString(1, "pk0001");
+      stmt.setString(2, "0002");
+      stmt.setObject(3, bsonDocument1);
+      stmt.executeUpdate();
+      conn.commit();
+
+      ResultSet rs = conn.createStatement()
+        .executeQuery("SELECT BSON_VALUE_TYPE(COL, 'non_existent_field') FROM " + tableName
+          + " WHERE PK1 = 'pk0001'");
+      assertTrue(rs.next());
+      assertEquals("NULL", rs.getString(1));
+
+      rs = conn.createStatement().executeQuery(
+        "SELECT BSON_VALUE_TYPE(COL, 'track[999]') FROM " + tableName + " WHERE PK1 = 'pk0001'");
+      assertTrue(rs.next());
+      assertEquals("NULL", rs.getString(1));
+
+      rs = conn.createStatement().executeQuery(
+        "SELECT BSON_VALUE_TYPE(COL, '') FROM " + tableName + " WHERE PK1 = 'pk0001'");
+      assertTrue(rs.next());
+      assertNull(rs.getString(1));
+
+      rs = conn.createStatement().executeQuery(
+        "SELECT BSON_VALUE_TYPE(COL, NULL) FROM " + tableName + " WHERE PK1 = 'pk0001'");
```

Comment:
Does this work? I thought `BSON_VALUE_TYPE(COL, 'NULL')` would be considered valid rather than `BSON_VALUE_TYPE(COL, NULL)`

Reply:
This is testing negative scenario. It returns NULL

Reply:
So it takes the argument as if nothing was provided?

Reply:
Right